src/syncerd/bitcoin_syncer.rs

```diff
@@ -242,20 +242,72 @@ impl ElectrumRpc {
     async fn query_transactions(&self, state: Arc<Mutex<SyncerState>>) {
         let state_guard = state.lock().await;
         let transactions = state_guard.transactions.clone();
+        let current_block_height = state_guard.block_height();
         drop(state_guard);
         for (_, watched_tx) in transactions.iter() {
             let tx_id = bitcoin::Txid::from_slice(&watched_tx.task.hash).unwrap();
-            match self.client.transaction_get_verbose(&tx_id) {
+            // Get the full transaction
+            match self.client.transaction_get(&tx_id) {
                 Ok(tx) => {
                     debug!("Updated tx: {}", &tx_id);
-                    let blockhash = tx.blockhash.map(|x| x.to_vec());
-                    let confs = match tx.confirmations {
-                        Some(confs) => confs,
-                        None => 0,
+                    // Look for history of the first output (maybe last is generally less likelly
+                    // to be used multiple times, so more efficient?!)
+                    let history = match self.client.script_get_history(&tx.output[0].script_pubkey)
+                    {
+                        Ok(history) => history,
+                        Err(err) => {
+                            debug!(
+                                "error getting script history, treating as not found: {}",
+                                err
+                            );
+                            let mut state_guard = state.lock().await;
+                            state_guard
+                                .change_transaction(tx_id.to_vec(), None, None)
```

Comment:
`script_get_history` errors if there is no confirmation yet, so this should be `change_transaction(tx_id.to_vec(), None, Some(0))`.

Reply:
Done